crates/starknet-devnet/tests/test_simulate_transactions.rs

```diff
@@ -232,6 +234,204 @@ mod simulation_tests {
+
+        update_gas_scenario(fork_devnet, "0x534e5f494e544547524154494f4e5f5345504f4c4941").await;
+    }
+
+    #[tokio::test]
+    async fn update_gas_check_blocks() {
+        let devnet = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+        devnet.create_block().await.unwrap();
+        let latest_block = devnet.get_latest_block_with_txs().await.unwrap();
+        assert_eq!(latest_block.block_number, 1);
+        assert_eq!(
+            latest_block.l1_gas_price.price_in_wei,
+            FieldElement::from(u128::from(DEVNET_DEFAULT_GAS_PRICE))
+        );
+        assert_eq!(
+            latest_block.l1_gas_price.price_in_fri,
+            FieldElement::from(u128::from(DEVNET_DEFAULT_GAS_PRICE))
+        );
+
+        let wei_price = 9e18 as u128;
+        let strk_price = 7e18 as u128;
+        let gas_update = json!({
+            "gas_price_wei": wei_price,
+            "data_gas_price_wei": 8e18 as u128,
+            "gas_price_strk": strk_price,
+            "data_gas_price_strk": 6e18 as u128,
+            "generate_block": false,
+        });
+        let updated_gas =
+            &devnet.send_custom_rpc("devnet_updateGas", gas_update.clone()).await.unwrap();
+        assert_eq!(updated_gas, &gas_update);
+
+        let latest_block = devnet.get_latest_block_with_txs().await.unwrap();
+        assert_eq!(latest_block.block_number, 1);
+
+        let pending_block = devnet.get_pending_block_with_tx_hashes().await.unwrap();
+        assert_eq!(pending_block.l1_gas_price.price_in_wei, FieldElement::from(wei_price));
+        assert_eq!(pending_block.l1_gas_price.price_in_fri, FieldElement::from(strk_price));
+
+        devnet.create_block().await.unwrap();
+
+        let latest_block = devnet.get_latest_block_with_txs().await.unwrap();
+        assert_eq!(latest_block.block_number, 2);
+        assert_eq!(latest_block.l1_gas_price.price_in_wei, FieldElement::from(wei_price));
+        assert_eq!(latest_block.l1_gas_price.price_in_fri, FieldElement::from(strk_price));
+
+        let wei_price = 8e18 as u128;
+        let strk_price = 6e18 as u128;
+        let gas_update_with_new_block = json!({
+            "gas_price_wei": wei_price,
+            "data_gas_price_wei": 7e18 as u128,
+            "gas_price_strk": strk_price,
+            "data_gas_price_strk": 5e18 as u128,
+            "generate_block": true,
+        });
+        let updated_gas = &devnet
+            .send_custom_rpc("devnet_updateGas", gas_update_with_new_block.clone())
+            .await
+            .unwrap();
+        assert_eq!(updated_gas, &gas_update_with_new_block);
+
+        let latest_block = devnet.get_latest_block_with_txs().await.unwrap();
+        assert_eq!(latest_block.block_number, 3);
+        assert_eq!(latest_block.l1_gas_price.price_in_wei, FieldElement::from(wei_price));
+        assert_eq!(latest_block.l1_gas_price.price_in_fri, FieldElement::from(strk_price));
```

Comment:
Here and in other places, why not do:
```rust
assert_eq!(latest_block.l1_gas_price, expected_object);
```

IMO it's always better to compare as many properties of an object as possible at the same time.

Reply:
Fixed in c5cf98b3c82352a9fc30c6763e8663302a51a5df